astropy/table/tests/test_column.py

```diff
@@ -464,6 +465,22 @@ def test_mask_on_non_masked_table(self):
             t["a"].mask = [True, False]
 
 
+@pytest.mark.parametrize(
+    "data",
+    [np.array([object()]), [object()]],
+)
+def test_deepcopy_object_column(data):
+    # see https://github.com/astropy/astropy/issues/13435
+    c1 = table.Column(data)
```

Comment:
Since we are here, you should define `meta = {1: object()}` and then ensure that the copy for `meta` is deep as well.